JustSaying.Extensions.DependencyInjection.Microsoft/IServiceCollectionExtensions.cs

```diff
@@ -199,6 +200,53 @@ public static IServiceCollection AddJustSaying(this IServiceCollection services,
             return services;
         }
 
+        /// <summary>
+        /// Adds a JustSaying message handler to the service collection that uses the specified handlers.
+        /// </summary>
+        /// <typeparam name="TMessage">The type of the message handled.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the message handler to.</param>
+        /// <param name="handlers">The message handlers to use.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/> specified by <paramref name="services"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> or <paramref name="handlers"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="handlers"/> contains no handlers.
+        /// </exception>
+        public static IServiceCollection AddJustSayingHandlers<TMessage>(
+            this IServiceCollection services,
+            IList<IHandlerAsync<TMessage>> handlers)
```

Comment:
I would go with an `IEnumerable<>` here because it's the least powerful and most widely useful interface, also it's what the constructor of `ListHandler` takes as parameter.
If it's enumerated multiple times and you want to avoid that, then convert it internally  with `.ToList()`.
If that's not acceptable, then maybe `IReadOnlyCollection<>` - just not a mutable list.

Reply:
I originally had it as `IEnumerable<T>`, but then I changed it to `ICollection<T>` so I could see if it was empty without having to use a LINQ `Count()` on it. Then I changed it to `IList<T>` so I could use the indexer to do the optimisation.

I can change it to `IReadOnly*<T>` if the distinction is important.

Reply:
If you're not going to modify it, then the `IReadOnly*` indicates that intent.